packages/crayons-extended/custom-objects/src/components/form-builder/utils/form-builder-utils.ts

```diff
@@ -275,3 +275,369 @@ export function checkIfCustomToggleField(
+        ...parentChoice.dependent_ids.choice.slice(0, delIndex),
+        ...parentChoice.dependent_ids.choice.slice(delIndex + 1),
+      ];
+    }
+  }
+
+  const depIds = choice.dependent_ids?.choice?.length
+    ? choice.dependent_ids.choice
+    : [];
+
+  // Delete current fields choices
+  const currentField = getFieldBasedOnLevel(field, level);
+  currentField.choices = deepCloneObject(
+    currentField.choices.filter((item) => item.id !== choice.id)
+  );
+
+  const deleteChildNodes = (depIds, currentLevel) => {
+    const childLevel = parseInt(currentLevel, 10) + 1;
+    const childField = getFieldBasedOnLevel(field, childLevel);
+    const childDepIds = [];
+    if (childField) {
+      childField.choices = deepCloneObject(
+        childField.choices.filter((item) => {
+          if (depIds.includes(item.id)) {
+            childDepIds.push(...item.dependent_ids.choice);
+            return false;
+          }
+
+          return true;
+        })
+      );
+    }
+
+    childDepIds.length && deleteChildNodes(childDepIds, childLevel);
+  };
+
+  depIds.length && deleteChildNodes(depIds, level);
+
+  return { ...field, fields: field.fields };
+}
+
+export function getDefaultDependentLevels(data, internalNamePrefix) {
+  const dataCloned = deepCloneObject(data);
+
+  function updateFieldAttribute(fProperties) {
+    fProperties.name = removeFirstOccurrence(
+      fProperties.name,
+      internalNamePrefix
+    );
+    fProperties.label = fProperties.label || '';
+    fProperties.id = fProperties.id || '';
+
+    if (hasCustomProperty(fProperties, 'fields') && fProperties.fields.length) {
+      updateFieldAttribute(fProperties.fields[0]);
+    }
+  }
+
+  updateFieldAttribute(dataCloned.fields[0]);
+
+  return { ...dataCloned, fields: dataCloned.fields };
+}
+
+export function addBulkChoices(data, text, { level, parentId, choiceIds }) {
+  const dataCloned = deepCloneObject(data);
+  const getField = getFieldBasedOnLevel(dataCloned, level);
+  const arrChoices = text.split('\n');
+  const seen = new Set();
+  let filteredChoices = getField.choices;
+  let parentField, parentChoice;
+
+  if (parentId) {
+    parentField = getFieldBasedOnLevel(dataCloned, parseInt(level, 10) - 1);
+    parentChoice = findChoice(parentField.choices, parentId);
+    filteredChoices = getField.choices.filter((item) =>
+      choiceIds.includes(item.id)
+    );
+  }
+
+  arrChoices.forEach((value) => {
+    value = value?.trim();
+    if (
+      value &&
+      value !== '' &&
+      !validateChoices(filteredChoices, value) &&
+      !seen.has(value)
+    ) {
+      const id = createUUID();
+      const choice = {
+        id,
+        value,
+        dependent_ids: { field: [], choice: [] },
+      };
+      getField.choices.push(choice);
+      if (parentChoice?.id) {
+        parentChoice.dependent_ids.choice.push(id);
+      }
+
+      // Adding temp to check the duplication and remove
+      seen.add(value);
+    }
+  });
+
+  return dataCloned;
+}
```

Comment:
Add UT's for util functions if possible please. And work on optimizations

Reply:
There is issue in writing tests, jest.mock is not allowed and global/Translation.ts is getting defined. Might require more time to add UT's. I will do this as part of code optimization in next PR.